src/back/spv/writer.rs

```diff
@@ -669,7 +669,7 @@ impl Writer {
         let pointer_id = self.get_pointer_id(arena, global_variable.ty, class);
         let instruction = super::instructions::instruction_variable(pointer_id, id, class, None);
 
-        if self.writer_flags.contains(WriterFlags::DEBUG) {
+        if self.writer_flags.contains(WriterFlags::DEBUG) && global_variable.name.is_some() {
```

Comment:
It would be better to avoid checking for this twice (first `is_some()`, then in `unwrap()`.
instead, we should be doing something like:
```rust
if self.writer_flags.contains(WriterFlags::DEBUG) {
  if let Some(ref name) = global_variable.name {
    ..
  }
}
```

Reply:
Sure, that sounds good! Update incoming.